src/operator/last.ts

```diff
@@ -16,14 +17,20 @@ import {EmptyError} from '../util/EmptyError';
  * from the source, or an NoSuchElementException if no such items are emitted.
  * @throws - Throws if no items that match the predicate are emitted by the source Observable.
  */
-export function last<T, R>(predicate?: (value: T, index: number, source: Observable<T>) => boolean,
-                           resultSelector?: (value: T, index: number) => R,
-                           defaultValue?: R): Observable<T> | Observable<R> {
+export function last<T, R>(predicate?: _sourcePredicate<T>,
+                           resultSelector?: (value: T, index: number) => R  | void,
+                           defaultValue?: R): Observable<T | R> {
   return this.lift(new LastOperator(predicate, resultSelector, defaultValue, this));
 }
 
+export interface LastSignature<T> {
+  (predicate?: _sourcePredicate<T>): Observable<T>;
+  (predicate: _sourcePredicate<T>, resultSelector: void, defaultValue?: T): Observable<T>;
```

Comment:
is `resultSelector` is just `void` in this case instead of function does not returns such as `(..) => void`? line 21 shows now there are 2 types of reselectselector and one is function type does not returns.